File: back-end-java/uiadmin-spring-boot/summer-core/src/main/java/com/jiangruyi/summer/core/entity/MenuItem.java
package com.jiangruyi.summer.core.entity;

import java.lang.Cloneable;
import java.io.Serializable;
import lombok.Data;

/**
 * 菜单实体类
 */
@Data
public class MenuItem implements Cloneable,Serializable {

    private static final long serialVersionUID = 1L;

    private String title;

    private String icon;

    private String path;

    private String pmenu;

    private String tip;

    private String menuLayer;

    private int menuType; // -1顶级文件夹0文件夹1菜单2按钮3接口

    private String routeType; // form/list/info/stack/tab

    private String apiPrefix;

    private String apiSuffix;

    private String apiParams;

    private String apiMethod;

    private String apiExt;

    private int isHide;

    private int status;

    private int sortnum;

    private String pathSuffix;

    private String outUrl;

    private String fullPath;
}

File: back-end-php/uiadmin-hyperf/appext/uiadmin-core/attributes/MenuItem.php
<?php
namespace uiadmin\core\attributes;

use Attribute;

/**
 * 注解菜单路由
 *
 * @author jry <ijry@qq.com>
 */
// 声明该Route注解可以重复使用，只能绑定在类方法上
#[Attribute(Attribute::IS_REPEATABLE|Attribute::TARGET_METHOD)]
class MenuItem {

    public static $all = [];

    public string $fullPath = '';
    public string $module = '';
    public string $title = '';
    public string $path = '';
    public string $pmenu = '';
    public string $tip = '';
    public string $menuLayer = 'admin';
    public int $menuType = 1;
    public string $routeType = 'form';
    public string $apiPrefix = 'v1';
    public string $apiSuffix = '';
    public string $apiParams = '';
    public string $apiMethod = 'GET';
    public string $apiExt = '';
    public int $isHide = 0;
    public int $status = 1;
    public int $sortnum = 0;
    public string $pathSuffix = '';
    public string $outUrl = '';

    public function __construct()
    {
    }

    /**
     * @description => 添加路由
     *
     *  @author jry <ijry@qq.com>
     * @return void
     */
    public function addRoute() :void
    {
        self::$all[] = [
            "fullPath" => $this->fullPath,
            "module" => $this->module,
            "title" => $this->title,
            "path" => $this->path,
            "pmenu" => $this->pmenu,
            "tip" => $this->tip,
            "menuLayer" => $this->menuLayer,
            "menuType" => $this->menuType,
            "routeType" => $this->routeType,
            "apiPrefix" => $this->apiPrefix,
            "apiSuffix" => $this->apiSuffix,
            "apiParams" => $this->apiParams,
            "apiMethod" => $this->apiMethod,
            "apiExt" => $this->apiExt,
            "isHide" => $this->isHide,
            "status" => $this->status,
            "sortnum" => $this->sortnum,
            "pathSuffix" => $this->pathSuffix,
            "outUrl" => $this->outUrl
        ];
    }
}
